src/include/net.h

```diff
@@ -21,6 +21,16 @@ int ncclNetVersion(struct ncclComm* comm);
 // Test whether the current GPU support GPU Direct RDMA.
 ncclResult_t ncclGpuGdrSupport(struct ncclComm* comm, int* gdrSupport);
 
+struct connectionMetaData_t {
+   bool send;
+   int tpRank;
+   int tpLocalRank;
+   int tpRemoteRank;
+   int channelId;
```

Comment:
If there are multiple connections per process, `connectionMetaData_t` might get overwritten. Why not have a `connectionMetaData` per `ncclProxyConnection`?